### ISSUE TITLE
[PeerConnector] bug with empty database

### ISSUE THREAD

**Issue by Aprogiena** [stratisproject/StratisBitcoinFullNode#1082]
got into this while using `-addnode` with the empty datadir

```
crit: Stratis.Bitcoin.FullNode[0]
      PeerConnectorDiscovery.ConnectAsync threw an unhandled exception
System.ArgumentOutOfRangeException: 'maxValue' must be greater than zero.
Parameter name: maxValue
   at System.Random.Next(Int32 maxValue)
   at Stratis.Bitcoin.P2P.PeerSelector.Random(IEnumerable`1 peers) in C:\Stratis\src\Stratis.Bitcoin\P2P\PeerSelector.cs:line 215
   at Stratis.Bitcoin.P2P.PeerSelector.SelectPeer() in C:\Stratis\src\Stratis.Bitcoin\P2P\PeerSelector.cs:line 73
   at Stratis.Bitcoin.P2P.PeerConnectorDiscovery.<OnConnectAsync>d__7.MoveNext() in C:\Stratis\src\Stratis.Bitcoin\P2P\PeerConnectorDiscovery.cs:line 80
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Stratis.Bitcoin.P2P.PeerConnector.<<StartConnectAsync>b__44_0>d.MoveNext() in C:\Stratis\src\Stratis.Bitcoin\P2P\PeerConnector.cs:line 222
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Stratis.Bitcoin.Utilities.AsyncLoop.<>c__DisplayClass16_0.<<StartAsync>b__0>d.MoveNext() in C:\Stratis\src\Stratis.Bitcoin\Utilities\AsyncLoop.cs:line 136
```

**Comment by zeptin** [comment]
I can also confirm that I get this frequently running integration tests on regtest

**Comment by fassadlr** [comment]
@Aprogiena I'm looking at this one. I suspect a threading issue.

**Comment by fassadlr** [comment]
@Aprogiena Actually my PR #1076 is supposed to fix this issue, we had static extension methods over the peer address list as well as not properly querying the concurrent dictionary. This caused a threading issue whereby we were passing an empty list to the method which returns a random peer from a list, thus causing a index out of range exception.

Maybe we can review that PR ;)

**Comment by fassadlr** [comment]
Closed with #1078
